sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncInitialSetupActivity.kt

```diff
@@ -71,12 +97,37 @@ class SyncInitialSetupActivity : DuckDuckGoActivity() {
 
     private fun processCommand(command: SyncInitialSetupViewModel.Command) {
         when (command) {
-            is SyncInitialSetupViewModel.Command.ShowMessage -> {
+            is ShowMessage -> {
                 Toast.makeText(this, command.message, Toast.LENGTH_LONG).show()
             }
+
+            ReadQR -> {
+                barcodeLauncher.launch(getScanOptions())
+            }
+
+            is ShowQR -> {
+                try {
+                    val barcodeEncoder = BarcodeEncoder()
+                    val bitmap: Bitmap = barcodeEncoder.encodeBitmap(command.string, QR_CODE, 400, 400)
```

Comment:
shouldn't this be done in a background thread?

Reply:
not sure 100%, imho we don't need to move into background here, but happy to change it if you see a reason. I will move forward with how it is now, but I can do it in the final screens.